project.rb

```diff
@@ -16,8 +17,18 @@ def remove_task(task_name)
     @tasks.reject! { |task| task.name == task_name }
   end
 
-  def to_s
+  def add_user(user, role)
+    @team << { user: user, role: role }
+  end
+
+  def to_s(user)
     task_list = @tasks.map(&:to_s).join("\n")
-    "Project: #{@name}\nTasks:\n#{task_list}"
+    "Project: #{@name}\nRole: #{user_role(user)}\nTasks:\n#{task_list}"
+  end
+
+  def user_role(user)
+    member = @team.select { |member| member[:user] == user.name }
```

Comment:
if you do `.find` instead of `.select` it will return the first one that matches (or `nil` if none match)